src/ccs/util/virtual_poly.rs

```diff
@@ -0,0 +1,498 @@
+use crate::hypercube::BooleanHypercube;
```

Comment:
We have a polynomial type in src/spartan/polynomial.rs, which was sufficient to implement the sum-check protocol for R1CS. I wonder if we can reuse that functionality rather than create an alternate type with overlapping functionality.